src/Eig/EigSolver.C

```diff
@@ -211,6 +225,19 @@ extern "C" void eig_sol_(const double* x, double* y, int& ierr)
   ierr = -99;
   if (!AM) return;
 
+#ifdef HAS_PETSC
+  const PETScMatrix* pA = dynamic_cast<const PETScMatrix*>(AM);
+  if (pA) {
+    PETScVector RHS(pA->getAdm(), x, AM->dim());
+    RHS.endAssembly();
+    ierr = AM->solve(RHS) ? 0 : -1;
+    PetscScalar* yv;
+    VecGetArray(RHS.getVector(), &yv);
+    memcpy(y,yv,RHS.dim()*sizeof(double));
```

Comment:
Same here. Need to release `yv`? Moreover, the memcpy will work only work if PetsCScalar and double are the same type (or at least same number of bytes per word). Is that always true? Otherwise, maybe std::copy would be more safe.

Reply:
Same reason. And yes petsc scalars are always double / we only support petsc with double scalars.

Reply:
Got it, PetscScalar=double, always